### PR TITLE
Cherry-pick #17391 to 7.x: [Agent] Rename agent to elastic-agent

### LINKED ISSUE #17239: [Agent] Rename our binary from agent to elastic-agent
The name "agent" is too generic to be installed on a system and will potentially conflict with other installed software we should rename the binaries to use the full name which is "elastic-agent" for the binary for every package generated.

This change needs to be done on the system/init.d files too, as a note heartbeat is using this strategy in the RPM/deb package.  The following is an extract of the debian package:

## Early list
- [ ] Change the directory x-pack/agent to x-pack/elastic-agent
- [ ] Change the binary to elastic-agent.
- [ ] Change the system.d / init.d to elastic-agent
- [ ] audit code  for "agent" mention to rename to Elastic Agent.
- [ ] Rename yml.

I think the above list is incomplete.

```
winterfell/tmp/out(:|✔) % tree
.
├── etc
│   ├── heartbeat
│   │   ├── fields.yml
│   │   ├── heartbeat.reference.yml
│   │   ├── heartbeat.yml
│   │   └── monitors.d
│   │       ├── sample.http.yml.disabled
│   │       ├── sample.icmp.yml.disabled
│   │       └── sample.tcp.yml.disabled
│   └── init.d
│       └── heartbeat-elastic
├── lib
│   └── systemd
│       └── system
│           └── heartbeat-elastic.service
└── usr
    ├── bin
    │   └── heartbeat
    └── share
        ├── doc
        │   └── heartbeat-elastic
        │       └── changelog.gz
        └── heartbeat
            ├── bin
            │   ├── heartbeat
            │   └── heartbeat-god
            ├── kibana
            ├── LICENSE.txt
            ├── NOTICE.txt
            └── README.md

### PR DESCRIPTION
**Pull request by ph**
Cherry-pick of PR elastic/beats#17391 to 7.x branch. Original message: 

## What does this PR do?

This PR renames agent to elastic-agent. 
packaging tested.
configuration variables such as `agent.id` are kept the same without a name change. 

## Why is it important?

To provide less generic name for executable

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

Fixes elastic/beats#17239

### CODE REVIEW

**Comment by ph** [comment]
@michalpristas don't merge it yet, I have another fix to go in.

**Comment by michalpristas** [comment]
Fails here: vet: management/fleet/manager.go:36:13: undeclared name: Config

**Comment by ph** [comment]
@michalpristas I've fixed all the bugs and diverging code from master to 7.x, I've run `make check` locally `mage package` and the tests. I think we are OK with this. Can you take another look?

**Comment by ph** [comment]
libbeat failures is related checking it out.

File: libbeat/cmd/instance/beat.go [hunk 1]
```diff
@@ -629,7 +629,7 @@ func (b *Beat) configure(settings Settings) error {
 	logp.Info("Beat ID: %v", b.Info.ID)
 
 	// initialize config manager
-	b.ConfigManager, err = management.Factory()(b.Config.Management, reload.Register, b.Beat.Info.ID)
```
Comment (michalpristas) [inline libbeat/cmd/instance/beat.go:632]: it's weird that this is missing

Comment (michalpristas) [reply libbeat/cmd/instance/beat.go:632]: `x-pack/libbeat/management/fleet` is missing entirely in branch

Comment (michalpristas) [reply libbeat/cmd/instance/beat.go:632]: check this PR and especially this test update to make libbeat pass
https://github.com/elastic/beats/pull/14835/files#diff-cf1cbef8b594350a8858a0682701458eR76

Comment (ph) [reply libbeat/cmd/instance/beat.go:632]: weird le met check that. I have it locally I might not have pushed it but I still have issues with the libbeat system test.

**Comment by ph** [comment]
@michalpristas I've went back to all the PRs I've backported and I did missed some files, so I tried to understand what happened and I just failed at the big merges. So I've think I've go everything now in this PR. Please take a look all tests were running locally will take a look at the CI this weekend.

**Pull request merged**